shared-lib/spiutils/src/protocol/flash.rs

```diff
@@ -221,15 +242,23 @@ pub struct Header<AddrType> {
 }
 
 impl<'a, AddrType> Header<AddrType>
-where AddrType: BeInt {
-    fn from_wire<R: Read<'a>>(mut r: R) -> Result<Self, FromWireError> {
+where AddrType: Address {
+    /// Get the address as an Option<u32>
+    pub fn get_address(&self) -> Option<u32> {
+        self.address.map(|val| val.into())
+    }
+
+    /// Deserializes a `Header` from `r`.
+    pub fn from_wire<R: Read<'a>>(mut r: R) -> Result<Self, FromWireError> {
         let opcode_u8 = r.read_be::<u8>()?;
         let opcode = OpCode::from_wire_value(opcode_u8).ok_or(FromWireError::OutOfRange)?;
 
-        let mut address: Option<AddrType> = None;
-        if opcode.has_address() {
-            address = Some(r.read_be::<AddrType>()?);
-        }
+        let address =
+            if opcode.has_address() {
+                Some(r.read_be::<AddrType>()?)
+            } else {
+                None
+            };
```

Comment:
This pattern can be written in 4 lines rather than 6 with `match`:

```
let address = match opcode.has_address() {
    true => Some(r.read_be::<AddrType>()?),
    false => None,
};
```

Reply:
One day ... one day I'll learn Rust!